kategory-core/src/main/kotlin/kategory/data/NonEmptyList.kt

```diff
@@ -25,7 +25,7 @@ class NonEmptyList<out A> private constructor(
 
     val size: Int = all.size
 
-    fun contains(element: @UnsafeVariance A): Boolean = (head == element).or(tail.contains(element))
+    fun contains(element: @UnsafeVariance A): Boolean = (head == element) || tail.contains(element)
```

Comment:
@kategory/maintainers This would avoid iterating through the tail if the value is in the head already.